packages/opencensus-core/src/trace/model/span.ts

```diff
@@ -310,10 +310,8 @@ export class Span implements types.Span {
       parentSpanId: this.parentSpanId,
       traceState: this.traceState
     });
-    if (!this.parentSpanId) {
-      this.tracer.setCurrentRootSpan(this);
-    }
 
+    if (this.isRootSpan()) this.tracer.setCurrentRootSpan(this);
```

Comment:
Why is `!this.parentSpanId` insufficient for determining that a span is a root span?

Reply:
A span, with a parent,  can be generated in one of two ways:
1) An outgoing call that's a child of an incoming request.  (Handled correctly)
2) An incoming call that's generated with trace and span id headers. (_Not_ handled correctly).

In case #2, the incoming span id is set to the parent id and outgoing calls are assigned a completely new trace id. It seems like a stop ship defect, tbh.

Reply:
OK, I understand now, in case #2 the span is not really a "root" span from the overall trace's perspective (the true root of the trace lies in an upstream service), but rather is only a "root" from this particular Node.js process's perspective.

In my opinion at some time in the future, it might be nice to just consolidate the distinction between a "root span" and a regular span as in this case it's actually incorrect to call it a root since it's only a root relative to this process. But I can approve this to unblock the issue.

Reply:
> In my opinion at some time in the future, it might be nice to just consolidate the distinction between a "root span" and a regular span as in this case it's actually incorrect to call it a root since it's only a root relative to this process. But I can approve this to unblock the issue.

I agree with you and I think we should consider this in OpenTelemetry project.